src/librustc/mir/mod.rs

```diff
@@ -2523,15 +2523,18 @@ impl<'tcx> Debug for Constant<'tcx> {
 
 impl<'tcx> Display for Constant<'tcx> {
     fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
+        use crate::ty::print::PrettyPrinter;
         write!(fmt, "const ")?;
-        // FIXME make the default pretty printing of raw pointers more detailed. Here we output the
-        // debug representation of raw pointers, so that the raw pointers in the mir dump output are
-        // detailed and just not '{pointer}'.
-        if let ty::RawPtr(_) = self.literal.ty.kind {
-            write!(fmt, "{:?} : {}", self.literal.val, self.literal.ty)
-        } else {
-            write!(fmt, "{}", self.literal)
-        }
+        ty::tls::with(|tcx| {
+            if let Some(literal) = tcx.lift(&self.literal) {
+                let mut cx = FmtPrinter::new(tcx, fmt, Namespace::ValueNS);
+                cx.print_alloc_ids = true;
+                cx.pretty_print_const(literal)?;
+                Ok(())
+            } else {
+                write!(fmt, "{}", self.literal)
```

Comment:
Pretty sure this will call `lift().unwrap()`? So you could just unwrap yourself above.